lib/inspec/utils/parser.rb

```diff
@@ -109,8 +121,16 @@ def parse_mount_options(mount_line, compatibility = false)
 
       # Device-/Sharename or Mountpoint includes whitespaces?
       def includes_whitespaces?(mount_line)
-        ws = mount_line.match(/^(.+)\son\s(.+)\stype\s.*$/)
-        ws.captures[0].include?(" ") || ws.captures[1].include?(" ")
+        # Split the mount_line by " on "
+        parts = mount_line.split(" on ")
+
+        # Check if the first part (device/share) or the second part (path) contains " type "
+        return false if parts.length != 2
```

Comment:
I guess you are assuming that if there is only 1 string then white space would not be present.. but the following has trailing white spaces.
```
'aa  on '.split(" on ")
=> ["aa "]
```

And why the parts should not be more than 2 ?

Reply:
examples of mount_line
`"//fileserver.corp.internal/Research & Development on /mnt/Research & Development type cifs (rw,vers=1.0)\n"`

`"/dev/xvda1 on / type ext4 (rw,discard)\n"`

Reply:
exactly, you could have the mount line with custom names, 
for example, see the below values and their behaviour
```
parts = "//fileserver.corp.internal/Research & Development on server   on /mnt/Research & Development type cifs (rw,vers=1.0)\n".split(" on ")
=> ["//fileserver.corp.internal/Research & Development", "server  ", "/mnt/Research & Development type cifs (rw,vers=1.0)\n"]
```